python/tskit/ms.py

```diff
@@ -0,0 +1,129 @@
+#
+# MIT License
+#
+# Copyright (c) 2019 Tskit Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+Convert tree sequences to ms output
+"""
+import numpy as np
+import sys
+import os
+
+from . import provenance
+
+class msWriter:
+    """
+    Writes a ms representation of the genotypes tree sequence to a
+    file-like object.
+    """
+
+    def __init__(
+        self,
+        tree_sequence,
+        print_trees=False,
+        precision=3,
+        mutation_rate=0,
+        recombination_rate=0,
+        migration_rate=0,
+        num_loci=1,
+        num_replicates=1,
+    ):
+        self._tree_sequence = tree_sequence
+        self._num_replicates = num_replicates
+        self._mutation_rate = mutation_rate
+        self._precision = precision
+        self._print_trees = print_trees
+        self._num_loci = num_loci
+
+    def print_trees(self, output):
+        """
+        Print out the trees in ms-format from the specified tree sequence.
+        When 'invisible' recombinations occur ms prints out copies of the
+        same tree. Therefore, we must keep track of all breakpoints from the
+        simulation and write out a tree for each one.
+        """
+        breakpoints = list(self._tree_sequence.breakpoints(True)) + [self._num_loci]
+        if self._num_loci == 1:
+            tree = next(self._tree_sequence.trees())
+            newick = tree.newick(precision=self._precision)
+            print(newick, file=output)
+        else:
+            j = 1
+            for tree in self._tree_sequence.trees():
```

Comment:
This can just be;
```
for tree in ts.trees():
    newick = tree.newick(precision=self._precision)
    print("[{}]".format(tree.span), newick, file=output)
```